Framework/PythonInterface/plugins/algorithms/ComputeIncoherentDOS.py

```diff
@@ -0,0 +1,196 @@
+
+        # Converts energy to meV for bose factor later
+        if u0 == 'DeltaE_inWavenumber' or u1 == 'DeltaE_inWavenumber':
+            en = en / mev2cm
+
+        # Checks energy bins are ok.
+        if EnergyBinning.count(',') != 2:
+            raise ValueError('EnergyBinning must be a comma separated string with three values.')
+        try:
+            ei = inws.getEFixed(1)
+        except RuntimeError:
+            ei = max(en)
+        try:
+            # Do this in a function to make sure no other variables can be evaluated other than Emin, Emax and Ei.
+            if cm:
+                dosebin = evaluateEbin(min(en*mev2cm), max(en*mev2cm), ei*mev2cm, EnergyBinning)
+            else:
+                dosebin = evaluateEbin(min(en), max(en), ei, EnergyBinning)
+        except NameError:
+            raise ValueError('Only the variables ''Emin'', ''Emax'' or ''Ei'' are allowed in EnergyBinning.')
+
+        # Extracts the intensity (y) and errors (e) from inws.
+        y = inws.extractY()
+        e = inws.extractE()
+        if iqq == 1:
+            y = np.transpose(y)
+            e = np.transpose(e)
+
+        # Creates a grid the same size as the intensity (y) array populated by the q and energy values
+        qqgrid = np.tile(np.array(qq), (np.shape(y)[0], 1))
+        engrid = np.transpose(np.tile(np.array(en), (np.shape(y)[1], 1)))
+
+        # Calculates the Debye-Waller and Bose factors from the Temperature and mean-squared displacements
+        DWF = np.exp(-2*(qqgrid**2*msd))
+        idm = np.where(engrid < 0)
+        idp = np.where(engrid >= 0)
+        expm = np.exp(-engrid[idm]*mev2k/Temperature)
+        expp = np.exp(-engrid[idp]*mev2k/Temperature)
+        Bose = DWF*0
+        Bose[idm] = expm / (1 - expm)      # n energy gain, phonon annihilation
+        Bose[idp] = expp / (1 - expp) + 1  # n energy loss, phonon creation
+        # Calculates the density of states from S(q,w)
+        y = y * (engrid/qqgrid**2) / (Bose*DWF)
+        e = e * (engrid/qqgrid**2) / (Bose*DWF)
+
+        # Checks if the user wants output in States/energy.
+        atoms, _ = inws.sample().getMaterial().chemicalFormula()
+        ylabel = 'g^{neutron}(E) (arb. units)'
+        if absunits:
+            if len(atoms) != 1:
+                self.log().warning('Sample material information not set, or sample is not a pure element. Ignoring StatesPerEnergy property.')
+                absunits = False
+            else:
+                if cm:
+                    ylabel = 'g(E) (states/cm^-1)'
+                else:
+                    ylabel = 'g(E) (states/meV)'
+
+        # Convert to wavenumbers if requested (y-axis is in mbarns/sr/fu/meV -> mb/sr/fu/cm^-1)
+        if cm:
+            en = en*mev2cm
+            y = y/mev2cm
+            e = e/mev2cm
+            yunit = 'DeltaE_inWavenumber'
+        else:
+            yunit = 'DeltaE'
+
+        # Outputs the calculated density of states to another workspace
```

Comment:
There is a problem here that causes a crash in debug mode on windows.  Your created workspace using CreateWorkspace is not setup correctly for Rebin2D to work on it.

The data on the old X edges is coming through as point data, not bin edges so the array overflows for the final boundary.

You will probably need to use the workspacefactory to create the workspace rather than CreateWorkspace to get one that is in the right form for Rebin2D.